src/main/java/org/blyznytsia/exception/CircularDependencyException.java

```diff
@@ -0,0 +1,11 @@
+package org.blyznytsia.exception;
+
+public class CircularDependencyException extends RuntimeException {
```

Comment:
Not sure if we need custom exceptions for each validation.
Having one `BeanValidationExcetion` should be enough.